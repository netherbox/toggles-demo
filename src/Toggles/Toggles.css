.Toggles {
  display: flex;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  padding: 20px;
}

.Toggles-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: linear-gradient(180deg, #f6b868 0%, #ee6b2d 100%);
  box-shadow: 0 0 10px #f6b868;
  border-radius: 20px;
  overflow: visible;
  padding: 20px;
}

.Toggles-title {
  text-align: center;
  font-style: normal;
  font-weight: bold;
  font-size: 40px;
  color: #fff;
  max-width: 900px;
  margin: 50px 0;
  width: 100%;
}

.Toggles-answers {
  display: flex;
  flex-direction: column;
  gap: 24px;
  font-size: 30px;
  max-width: 900px;
  margin: 20px 0;
  width: 100%;
}

.Toggles-result {
  text-align: center;
  font: normal;
  font-weight: bold;
  color: #fff;
  font-size: 32px;
  max-width: 900px;
  margin: 50px 0;
  width: 100%;
}

@media (max-width: 1200px) {
  .Toggles-container {
    padding: 20px;
  }

  .Toggles-title {
    font-size: 35px;
    margin: 40px 0;
  }

  .Toggles-answers {
    font-size: 25px;
    margin: 20px 0;
    gap: 20px;
  }

  .Toggles-result {
    font-size: 35px;
    margin: 40px 0;
  }
}

@media (max-width: 992px) {
  .Toggles-container {
    padding: 17px;
  }

  .Toggles-title {
    font-size: 25px;
    margin: 30px 0;
  }

  .Toggles-answers {
    font-size: 20px;
    margin: 17px 0;
    gap: 15px;
  }

  .Toggles-result {
    font-size: 25px;
    margin: 30px 0;
  }
}

@media (max-width: 768px) {
  .Toggles-container {
    padding: 17px;
  }

  .Toggles-title {
    font-size: 22px;
    margin: 25px 0;
  }

  .Toggles-answers {
    font-size: 20px;
    margin: 17px 0;
    gap: 24px;
  }

  .Toggles-result {
    font-size: 22px;
    margin: 25px 0;
  }
}

@media (max-width: 576px) {
  .Toggles {
    padding: 0;
    box-shadow: none;
  }

  .Toggles-container {
    padding: 17px;
    min-width: 320px;
    border-radius: 0;
  }

  .Toggles-title {
    font-size: 20px;
    margin: 20px 0;
  }

  .Toggles-answers {
    font-size: 18px;
    margin: 15px 0;
    gap: 32px;
  }

  .Toggles-result {
    font-size: 20px;
    margin: 30px 0;
  }
}
