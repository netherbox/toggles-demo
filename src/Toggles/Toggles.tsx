import { Component } from "react";
import Toggle, { Answer } from "../Toggle/Toggle";
import shuffle from "lodash/shuffle";
import random from "lodash/random";

import "./Toggles.css";

export interface Question {
  id: string;
  answers: Array<Answer>;
  selectedAnswerId: string;
  correctAnswerId: string;
}

export interface TogglesPropsQuestion {
  id: string;
  answers: Array<Answer>;
  correctAnswerId: string;
}

export interface TogglesProps {
  id: string;
  title: string;
  questions: Array<TogglesPropsQuestion>;
  disableShuffle?: boolean;
}

export interface TogglesState {
  questions: Array<Question>;
  isCorrect: boolean;
  gradientTopColor: string;
  gradientBottomColor: string;
  borderColor: string;
  answerColor: string;
  selectedAnswerColor: string;
  sliderColor: string;
}

export class Toggles extends Component<TogglesProps, TogglesState> {
  // Color steps order from incorreect -> correct
  theme = {
    gradientTopColor: ["#F1B496", "#F6B868", "#76E0C2"],
    gradientBottomColor: ["#EA806A", "#EE6B2D", "#59CADA"],
    textColor: ["#FFFFFF", "#FFFFFF", "#FFFFFF"],
    borderColor: ["#FBFBFB", "#F9D29F", "#FBFBFB"],
    answerColor: ["#FFFFFF", "#FFFFFF", "#FFFFFF"],
    selectedAnswerColor: ["#E47958", "#9F938B", "#4CAD94"],
    sliderColor: ["#F2CBBD", "#F8CAA3", "#A5E7E2"],
  };

  getClosestColor(colors: Array<string>, position: number) {
    return colors[Math.floor((colors.length - 1) * position)];
  }

  constructor(props: TogglesProps) {
    super(props);

    let questions: Array<Question>;

    if (this.props.disableShuffle) {
      // If shuffle is disabled set the selected answer to the first one
      questions = this.props.questions.map(
        (question: TogglesPropsQuestion) => ({
          ...question,
          answers: question.answers,
          selectedAnswerId: question.answers[0].id,
        })
      );
    } else {
      // Shuffle the questions, answers and set random current selected answer
      questions = shuffle(
        this.props.questions.map((question: TogglesPropsQuestion) => ({
          ...question,
          answers: shuffle(question.answers),
          selectedAnswerId:
            question.answers[random(0, question.answers.length - 1)].id,
        }))
      );
    }

    this.state = {
      questions,
      isCorrect: this.isCorrect(questions),
      ...this.getCorrectnessTheme(questions),
    };
  }

  isCorrect(questions: Array<Question>): boolean {
    return questions.reduce(
      (accumulator: boolean, question: Question) =>
        accumulator && question.correctAnswerId === question.selectedAnswerId,
      true
    );
  }

  getCorrectness(questions: Array<Question>): number {
    return (
      questions.reduce(
        (accumulator: number, question: Question) =>
          accumulator +
          (question.correctAnswerId === question.selectedAnswerId ? 1 : 0),
        0
      ) / questions.length
    );
  }

  private getCorrectnessTheme(questions: Array<Question>) {
    const correctness = this.getCorrectness(questions);

    return {
      gradientTopColor: this.getClosestColor(
        this.theme.gradientTopColor,
        correctness
      ),
      gradientBottomColor: this.getClosestColor(
        this.theme.gradientBottomColor,
        correctness
      ),
      borderColor: this.getClosestColor(this.theme.borderColor, correctness),
      answerColor: this.getClosestColor(this.theme.answerColor, correctness),
      selectedAnswerColor: this.getClosestColor(
        this.theme.selectedAnswerColor,
        correctness
      ),
      sliderColor: this.getClosestColor(this.theme.sliderColor, correctness),
    };
  }

  handleChange(questionId: string, selectedAnswerId: string) {
    this.setState((prevState: TogglesState) => {
      const questions = prevState.questions.map((question: Question) => ({
        ...question,
        selectedAnswerId:
          question.id === questionId
            ? selectedAnswerId
            : question.selectedAnswerId,
      }));

      return {
        questions,
        isCorrect: this.isCorrect(questions),
        ...this.getCorrectnessTheme(questions),
      };
    });
  }

  getContainerStyle() {
    return {
      background: `linear-gradient(180deg, ${this.state.gradientTopColor} 0%, ${this.state.gradientBottomColor} 100%)`,
      boxShadow: `0 0 10px ${this.state.gradientTopColor}`,
    };
  }

  render() {
    const toggleItems = this.state.questions.map((question: Question) => (
      <Toggle
        id={`${this.props.id}-${question.id}`}
        key={question.id}
        answers={question.answers}
        selectedAnswerId={question.selectedAnswerId}
        correctAnswerId={question.correctAnswerId}
        locked={this.state.isCorrect}
        borderColor={this.state.borderColor}
        answerColor={this.state.answerColor}
        selectedAnswerColor={this.state.selectedAnswerColor}
        sliderColor={this.state.sliderColor}
        onChange={(selectedAnswerId) =>
          this.handleChange(question.id, selectedAnswerId)
        }
      ></Toggle>
    ));

    return (
      <div className="Toggles">
        <div
          className="Toggles-container"
          data-testid="toggles-container"
          style={this.getContainerStyle()}
        >
          <div className="Toggles-title">{this.props.title}</div>
          <div className="Toggles-answers">{toggleItems}</div>
          <div className="Toggles-result">
            {this.state.isCorrect
              ? "The answer is correct!"
              : "The answer is incorrect"}
          </div>
        </div>
      </div>
    );
  }
}

export default Toggles;
