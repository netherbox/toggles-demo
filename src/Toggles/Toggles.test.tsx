import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import Toggles from "./Toggles";

describe("Should show correct/incorrect message", () => {
  it("should display is correct message if answer is correct", async () => {
    render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a1", text: "Valid answer" },
              { id: "a2", text: "Invalid answer" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(screen.getByText("The answer is correct!")).toBeInTheDocument();
  });

  it("should display is incorrect message if answer is incorrect", async () => {
    render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a2", text: "Invalid answer" },
              { id: "a1", text: "Valid answer" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(screen.getByText("The answer is incorrect")).toBeInTheDocument();
  });
});

describe("The solution should lock once the correct answer is reached so the toggles can no longer be switched", () => {
  it("should allow to change answer if incorrect", async () => {
    render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a2", text: "Invalid answer" },
              { id: "a1", text: "Valid answer" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(screen.getByText("The answer is incorrect")).toBeInTheDocument();

    fireEvent.click(screen.getByText("Valid answer"));

    await waitFor(() => {
      expect(screen.getByText("The answer is correct!")).toBeInTheDocument();
    });
  });

  it("should not allow to change answer if correct", async () => {
    render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a1", text: "Valid answer" },
              { id: "a2", text: "Invalid answer" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(screen.getByText("The answer is correct!")).toBeInTheDocument();

    fireEvent.click(screen.getByText("Invalid answer"));

    await waitFor(() => {
      expect(screen.getByText("The answer is correct!")).toBeInTheDocument();
    });
  });
});

describe('The background color should change in proportion to how "correct" the answer is', () => {
  it("should set the colors to red if all three answers are incorrect", async () => {
    const { container } = render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a2", text: "Invalid answer 1" },
              { id: "a1", text: "Valid answer 1" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q2",
            answers: [
              { id: "a2", text: "Invalid answer 2" },
              { id: "a1", text: "Valid answer 2" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q3",
            answers: [
              { id: "a2", text: "Invalid answer 3" },
              { id: "a1", text: "Valid answer 3" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should set the colors to red if one of the answers in incorrect", async () => {
    const { container } = render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
                { id: "a1", text: "Valid answer 1" },
                { id: "a2", text: "Invalid answer 1" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q2",
            answers: [
              { id: "a2", text: "Invalid answer 2" },
              { id: "a1", text: "Valid answer 2" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q3",
            answers: [
              { id: "a2", text: "Invalid answer 3" },
              { id: "a1", text: "Valid answer 3" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should set the colors to orange if two of the answers in incorrect", async () => {
    const { container } = render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a1", text: "Valid answer 1" },
              { id: "a2", text: "Invalid answer 1" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q2",
            answers: [
              { id: "a1", text: "Valid answer 2" },
              { id: "a2", text: "Invalid answer 2" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q3",
            answers: [
              { id: "a2", text: "Invalid answer 3" },
              { id: "a1", text: "Valid answer 3" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should set the colors to green if all three answers are correct", async () => {
    const { container } = render(
      <Toggles
        id="t1"
        title="Really easy question:"
        questions={[
          {
            id: "q1",
            answers: [
              { id: "a1", text: "Valid answer 1" },
              { id: "a2", text: "Invalid answer 1" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q2",
            answers: [
              { id: "a1", text: "Valid answer 2" },
              { id: "a2", text: "Invalid answer 2" },
            ],
            correctAnswerId: "a1",
          },
          {
            id: "q3",
            answers: [
              { id: "a1", text: "Valid answer 3" },
              { id: "a2", text: "Invalid answer 3" },
            ],
            correctAnswerId: "a1",
          },
        ]}
        disableShuffle={true}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
