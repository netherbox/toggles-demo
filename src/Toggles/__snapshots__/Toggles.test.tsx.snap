// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The background color should change in proportion to how "correct" the answer is should set the colors to green if all three answers are correct 1`] = `
<div
  class="Toggles"
>
  <div
    class="Toggles-container"
    data-testid="toggles-container"
    style="box-shadow: 0 0 10px #76E0C2;"
  >
    <div
      class="Toggles-title"
    >
      Really easy question:
    </div>
    <div
      class="Toggles-answers"
    >
      <div
        class="Toggle locked"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(165, 231, 226); border-color: #a5e7e2;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            disabled=""
            id="t1-q1-a1"
            name="t1-q1"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q1-a1"
            style="color: rgb(76, 173, 148);"
            tabindex="-1"
          >
            Valid answer 1
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            disabled=""
            id="t1-q1-a2"
            name="t1-q1"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q1-a2"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Invalid answer 1
          </label>
        </div>
      </div>
      <div
        class="Toggle locked"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(165, 231, 226); border-color: #a5e7e2;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            disabled=""
            id="t1-q2-a1"
            name="t1-q2"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q2-a1"
            style="color: rgb(76, 173, 148);"
            tabindex="-1"
          >
            Valid answer 2
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            disabled=""
            id="t1-q2-a2"
            name="t1-q2"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q2-a2"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Invalid answer 2
          </label>
        </div>
      </div>
      <div
        class="Toggle locked"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(165, 231, 226); border-color: #a5e7e2;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            disabled=""
            id="t1-q3-a1"
            name="t1-q3"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q3-a1"
            style="color: rgb(76, 173, 148);"
            tabindex="-1"
          >
            Valid answer 3
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            disabled=""
            id="t1-q3-a2"
            name="t1-q3"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q3-a2"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Invalid answer 3
          </label>
        </div>
      </div>
    </div>
    <div
      class="Toggles-result"
    >
      The answer is correct!
    </div>
  </div>
</div>
`;

exports[`The background color should change in proportion to how "correct" the answer is should set the colors to orange if two of the answers in incorrect 1`] = `
<div
  class="Toggles"
>
  <div
    class="Toggles-container"
    data-testid="toggles-container"
    style="box-shadow: 0 0 10px #F6B868;"
  >
    <div
      class="Toggles-title"
    >
      Really easy question:
    </div>
    <div
      class="Toggles-answers"
    >
      <div
        class="Toggle"
        style="border-color: #f9d29f;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(248, 202, 163); border-color: #f8caa3;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q1-a1"
            name="t1-q1"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q1-a1"
            style="color: rgb(159, 147, 139);"
            tabindex="-1"
          >
            Valid answer 1
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q1-a2"
            name="t1-q1"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q1-a2"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Invalid answer 1
          </label>
        </div>
      </div>
      <div
        class="Toggle"
        style="border-color: #f9d29f;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(248, 202, 163); border-color: #f8caa3;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q2-a1"
            name="t1-q2"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q2-a1"
            style="color: rgb(159, 147, 139);"
            tabindex="-1"
          >
            Valid answer 2
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q2-a2"
            name="t1-q2"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q2-a2"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Invalid answer 2
          </label>
        </div>
      </div>
      <div
        class="Toggle"
        style="border-color: #f9d29f;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(248, 202, 163); border-color: #f8caa3;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q3-a2"
            name="t1-q3"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q3-a2"
            style="color: rgb(159, 147, 139);"
            tabindex="-1"
          >
            Invalid answer 3
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q3-a1"
            name="t1-q3"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q3-a1"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Valid answer 3
          </label>
        </div>
      </div>
    </div>
    <div
      class="Toggles-result"
    >
      The answer is incorrect
    </div>
  </div>
</div>
`;

exports[`The background color should change in proportion to how "correct" the answer is should set the colors to red if all three answers are incorrect 1`] = `
<div
  class="Toggles"
>
  <div
    class="Toggles-container"
    data-testid="toggles-container"
    style="box-shadow: 0 0 10px #F1B496;"
  >
    <div
      class="Toggles-title"
    >
      Really easy question:
    </div>
    <div
      class="Toggles-answers"
    >
      <div
        class="Toggle"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(242, 203, 189); border-color: #f2cbbd;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q1-a2"
            name="t1-q1"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q1-a2"
            style="color: rgb(228, 121, 88);"
            tabindex="-1"
          >
            Invalid answer 1
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q1-a1"
            name="t1-q1"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q1-a1"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Valid answer 1
          </label>
        </div>
      </div>
      <div
        class="Toggle"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(242, 203, 189); border-color: #f2cbbd;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q2-a2"
            name="t1-q2"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q2-a2"
            style="color: rgb(228, 121, 88);"
            tabindex="-1"
          >
            Invalid answer 2
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q2-a1"
            name="t1-q2"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q2-a1"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Valid answer 2
          </label>
        </div>
      </div>
      <div
        class="Toggle"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(242, 203, 189); border-color: #f2cbbd;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q3-a2"
            name="t1-q3"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q3-a2"
            style="color: rgb(228, 121, 88);"
            tabindex="-1"
          >
            Invalid answer 3
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q3-a1"
            name="t1-q3"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q3-a1"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Valid answer 3
          </label>
        </div>
      </div>
    </div>
    <div
      class="Toggles-result"
    >
      The answer is incorrect
    </div>
  </div>
</div>
`;

exports[`The background color should change in proportion to how "correct" the answer is should set the colors to red if one of the answers in incorrect 1`] = `
<div
  class="Toggles"
>
  <div
    class="Toggles-container"
    data-testid="toggles-container"
    style="box-shadow: 0 0 10px #F1B496;"
  >
    <div
      class="Toggles-title"
    >
      Really easy question:
    </div>
    <div
      class="Toggles-answers"
    >
      <div
        class="Toggle"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(242, 203, 189); border-color: #f2cbbd;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q1-a1"
            name="t1-q1"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q1-a1"
            style="color: rgb(228, 121, 88);"
            tabindex="-1"
          >
            Valid answer 1
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q1-a2"
            name="t1-q1"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q1-a2"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Invalid answer 1
          </label>
        </div>
      </div>
      <div
        class="Toggle"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(242, 203, 189); border-color: #f2cbbd;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q2-a2"
            name="t1-q2"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q2-a2"
            style="color: rgb(228, 121, 88);"
            tabindex="-1"
          >
            Invalid answer 2
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q2-a1"
            name="t1-q2"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q2-a1"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Valid answer 2
          </label>
        </div>
      </div>
      <div
        class="Toggle"
        style="border-color: #fbfbfb;"
      >
        <div
          class="Toggle-slider"
          data-testid="slider"
          style="left: 0%; width: 50%; background-color: rgb(242, 203, 189); border-color: #f2cbbd;"
        />
        <div
          class="Toggle-answer selected"
          style="width: 50%;"
        >
          <input
            checked=""
            id="t1-q3-a2"
            name="t1-q3"
            type="radio"
            value="a2"
          />
          <label
            for="t1-q3-a2"
            style="color: rgb(228, 121, 88);"
            tabindex="-1"
          >
            Invalid answer 3
          </label>
        </div>
        <div
          class="Toggle-answer"
          style="width: 50%;"
        >
          <input
            id="t1-q3-a1"
            name="t1-q3"
            type="radio"
            value="a1"
          />
          <label
            for="t1-q3-a1"
            style="color: rgb(255, 255, 255);"
            tabindex="-1"
          >
            Valid answer 3
          </label>
        </div>
      </div>
    </div>
    <div
      class="Toggles-result"
    >
      The answer is incorrect
    </div>
  </div>
</div>
`;
